### PR TITLE
add question backend

### PR DESCRIPTION
**Pull request by JanhaviDadhania**
add practice question backend #34

go to temporary admin_login.php
type 'user' and submit
user will be directed to add_question.php page

checks:

- if any field is empty, it shows message to fill all fields.
- if all details are filled it shows message "successfully added" (wait for 4 seconds and user will be - redirected to main page)
- if question id matches with any other question, it shows message and redirect to main page

updates:

- it updates all testcases and all questions directories
- it updates practice_question table
- it creates new table for added question.

"all testcases" and "all questions" directories should have required permissions to create directory and add files
*update login.php before

### CODE REVIEW

File: CSE-Hub/admin/add_question.php [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+            </p>
+
+            <div class="container">
+                <form action = "update.php" method = "post">
+                    <ul class="add">
+                    <li class="add">Title: <input type="text" name="title"></li><br/>
+                    <li class="add">Question Id: <input type="text" name="ques_id"></li><br/><br/>
+                    <li class="add">
+                    Question Description: <br/><br/>
+                                      <textarea rows = "5" cols = "50" name = "description">
```
Comment (harshraj22) [inline CSE-Hub/admin/add_question.php]: Indentation ??

**Comment by harshraj22** [comment]
* Check if all fields are filled on front-end (in case of issue, [this](https://stackoverflow.com/questions/17061436/textareas-required-attribute-doesnt-work-even-though-the-value-is-empty) might help ).
* Currently the testcase that is saved contains unexpected tabs at the start of line.
Also adding multiple testcase option might be more helpful.

**Comment by JanhaviDadhania** [comment]
- added checks for all in html form (front end)
- indentation and tabs issue solved in previous commit
- added feature to add multiple testcases (at least one ... at most five)

**Comment by harshraj22** [comment]
Screenshot from 2019-10-08 10-30-18

Seems this needs to be fixed as well.

File: CSE-Hub/admin/add_question.php [hunk 2]
```diff
@@ -78,22 +78,56 @@
             <div class="container">
                 <form action = "update.php" method = "post">
                     <ul class="add">
-                    <li class="add">Title: <input type="text" name="title"></li><br/>
-                    <li class="add">Question Id: <input type="text" name="ques_id"></li><br/><br/>
+                    <li class="add">Title: <input type="text" name="title" required></li><br/>
+                    <li class="add">Question Id: <input type="text" name="ques_id" required></li><br/><br/>
+                    <li class="add">Date<input type="date" name="date" required></li><br/>
+                    <li class="add">Time Limit in seconds:<input type="number" name="time_limit" required></li><br/>
```
Comment (harshraj22) [inline CSE-Hub/admin/add_question.php]: Also, here adding min="0" step="0.5" might be better

**Comment by DestroyerAlpha** [comment]
Is the footer issue fixed? It doesn't seem fixed for me.

**Comment by DestroyerAlpha** [comment]
Issue
Please check

**Comment by DestroyerAlpha** [comment]
Also modify the Error page not found to 404 instead of 4040

**Comment by JanhaviDadhania** [comment]
added feature to add more test cases.
Test cases and solutions are stored in the form... ques_id1.txt, sol1.txt and so on...

error page modified.
modified footer styling.

File: CSE-Hub/admin/update.php [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    //adding description, testcases and solution text into files
+    file_put_contents("../practice questions/all questions/$ques_id/$name", $description);
+    file_put_contents("../practice questions/all testcases/$ques_id/solutions/sol.txt", $solution);
+    file_put_contents("../practice questions/all testcases/$ques_id/testcases/test.txt", $testcase);
+
+    // updating practice_question_info table
+    $newquery = 'INSERT INTO practice_questions_info (ques_ID, date_created, successful_submissions, total_submissions, admin) VALUES (?, ?, "0", "0", ?)';
+    prepared_query($conn, $newquery, [$ques_id, $date, $username]);
+
+    if(!$newquery)
```
Comment (harshraj22) [inline CSE-Hub/admin/update.php:70]: @DestroyerAlpha, @JanhaviDadhania  , Should we check $newquery or $stmt here ? As the result of query execution is stored in $stmt. I think this was missed in the other pr as well. Can you check it out ?

Comment (DestroyerAlpha) [reply CSE-Hub/admin/update.php:70]: I think a variable needs to be assigned to the function call.
Need to check though.

File: CSE-Hub/practice questions/all questions/q1/q1.txt [hunk 4]
```diff
@@ -1,27 +1,56 @@
...
+Subtask #2 (70 points): original constraints
+
+Example Input
+1
+4 2
+1 2 3 4
+Example Output
+1
+Explanation
+Example case 1: There are six subsequences with length 2: (1,2), (1,3), (1,4), (2,3), (2,4) and (3,4). The minimum sum is 3 and the only subsequence with this sum is (1,2).
```
Comment (harshraj22) [inline CSE-Hub/practice questions/all questions/q1/q1.txt]: Things seem repeated here.

**Pull request merged**